domain/src/Aggregates/SharePoolingAsset/Services/DisposalBuilder/DisposalBuilder.php

```diff
@@ -145,35 +147,14 @@ private static function processAcquisitionsWithinThirtyDays(
         foreach ($withinThirtyDaysAcquisitions as $acquisition) {
             // Apply the acquisition's cost basis to the disposed of asset up to the remaining quantity
             $thirtyDayQuantityToApply = Quantity::minimum($acquisition->availableThirtyDayQuantity(), $remainingQuantity);
-
-            // Also deduct same-day disposals with available same-day quantity that haven't been processed yet
-            $sameDayDisposals = $transactions->disposalsMadeOn($acquisition->date)
-                ->unprocessed()
-                ->withAvailableSameDayQuantity();
-
-            foreach ($sameDayDisposals as $disposal) {
-                $sameDayQuantityToApply = Quantity::minimum($disposal->availableSameDayQuantity(), $thirtyDayQuantityToApply);
-                $disposal->sameDayQuantityAllocation->allocateQuantity($sameDayQuantityToApply, $acquisition);
-                $acquisition->increaseSameDayQuantity($sameDayQuantityToApply);
-                $thirtyDayQuantityToApply = $thirtyDayQuantityToApply->minus($sameDayQuantityToApply);
-                if ($thirtyDayQuantityToApply->isZero()) {
-                    break;
-                }
-            }
-
-            if ($thirtyDayQuantityToApply->isZero()) {
-                continue;
-            }
```

Comment:
ℹ️ Disposals are not being added as unprocessed anymore, so this section is now redundant